hack/generate-operator-bundle.py

```diff
@@ -4,88 +4,172 @@
+    generate_package(args.channel, version)
+
+
+def generate_csv_osd():
+    print("Generating CSV for OpenShift Dedicated")
+    version = generate_osd_version()
+    generate_csv_base(version, args.prev_version, args.hive_image)
+    generate_package(args.channel, version)
+
+
+def generate_csv_base(version, prev_version, hive_image):
+    print("Generating CSV for version: %s" % version)
+
+    crds_dir = 'config/crds'
+    csv_template = 'config/templates/hive-csv-template.yaml'
+    operator_role = 'config/operator/operator_role.yaml'
+    deployment_spec = 'config/operator/operator_deployment.yaml'
+
+    if not os.path.exists(BUNDLE_DIR):
+        os.mkdir(BUNDLE_DIR)
+
+    version_dir = os.path.join(BUNDLE_DIR, version)
+    if not os.path.exists(version_dir):
+        os.mkdir(version_dir)
+
+    # Copy all CSV files over to the bundle output dir:
+    crd_files = os.listdir(crds_dir)
+    for file_name in crd_files:
+        full_path = os.path.join(crds_dir, file_name)
+        if os.path.isfile(os.path.join(crds_dir, file_name)):
+            shutil.copy(full_path, os.path.join(version_dir, file_name))
+
+    with open(csv_template, 'r') as stream:
+        csv = yaml.load(stream, Loader=yaml.SafeLoader)
+
+    csv['spec']['install']['spec']['clusterPermissions'] = []
+
+    # Add our operator role to the CSV:
+    with open(operator_role, 'r') as stream:
+        operator_role = yaml.load(stream, Loader=yaml.SafeLoader)
+        csv['spec']['install']['spec']['clusterPermissions'].append(
+            {
+                'rules': operator_role['rules'],
+                'serviceAccountName': 'hive-operator',
+            })
+
+    # Add our deployment spec for the hive operator:
+    with open(deployment_spec, 'r') as stream:
+        operator_components = []
+        operator = yaml.load_all(stream, Loader=yaml.SafeLoader)
+        for doc in operator:
+            operator_components.append(doc)
+        operator_deployment = operator_components[1]
+        csv['spec']['install']['spec']['deployments'][0]['spec'] = operator_deployment['spec']
+
+    # Update the versions to include git hash:
+    csv['metadata']['name'] = "hive-operator.v%s" % version
+    csv['spec']['version'] = version
+    csv['spec']['replaces'] = "hive-operator.v%s" % prev_version
+
+    # Update the deployment to use the defined image:
+    image_ref = hive_image
+    if not ":" in image_ref:
+        image_ref = "%s:v%s" % (hive_image, version)
+    csv['spec']['install']['spec']['deployments'][0]['spec']['template']['spec']['containers'][0]['image'] = image_ref
+    csv['metadata']['annotations']['containerImage'] = image_ref
+
+    # Set the CSV createdAt annotation:
+    now = datetime.datetime.now()
+    csv['metadata']['annotations']['createdAt'] = now.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # Write the CSV to disk:
+    csv_filename = "hive-operator.v%s.clusterserviceversion.yaml" % version
+    csv_file = os.path.join(version_dir, csv_filename)
```

Comment:
So will OLM manage the CRD's or Hive still? If you are putting them in the bundle that means that OLM is going to deploy them? Don't you want to add them to `csv['spec']['customresourcedefinitions']['owned']` ?